File: src/web/Providers/Models/GenerateScheduleRequest.cs
using Web.Providers.Models.Enums;
using Web.Providers.Models.Tasks;
using Web.Providers.Schedule.Models.Payload;
using Web.Providers.Schedule.Models.Tasks;

namespace Web.Providers.Models
{
    public record GenerateScheduleRequest
    {
        public required IReadOnlyList<FixedTask> FixedTasks { get; init; }
        public required IReadOnlyList<DynamicTask> DynamicTasks { get; init; }
        public required PlanningHorizon PlanningHorizon { get; init; }
        public IReadOnlyList<CategoryWindow> CategoryWindows { get; init; } = [];
        public required DifficultTaskSchedulingStrategy DifficultTaskSchedulingStrategy { get; init; }
        public IReadOnlyList<DifficultyCapacityEntry> DifficultyCapacities { get; init; } = [];
        public IReadOnlyList<TaskTypePreferenceEntry> TaskTypePreferences { get; init; } = [];
    }
}

File: src/timefold/src/main/java/com/pso/timefold/dto/GenerateScheduleRequest.java
package com.pso.timefold.dto;

import com.pso.timefold.domain.*;
import com.pso.timefold.domain.enums.DifficultTaskSchedulingStrategy;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

/**
 * JSON deserialization target for POST /jobs/run.
 * Mirrors web/Providers/Models/GenerateScheduleRequest (camelCase fields).
 */
@Getter
@Setter
@NoArgsConstructor
public class GenerateScheduleRequest {

    private List<FixedTask> fixedTasks;
    private List<DynamicTask> dynamicTasks;
    private PlanningHorizon planningHorizon;
    private List<CategoryWindow> categoryWindows;
    private DifficultTaskSchedulingStrategy difficultTaskSchedulingStrategy;
    private List<DifficultyCapacityEntry> difficultyCapacities;
    private List<TaskTypePreferenceEntry> taskTypePreferences;
    private int optimizationTimeInSeconds = 15;
}
